### PR TITLE
Introduce new overloads (mainly xxx3C)

### PR DESCRIPTION
**Pull request by sledorze**
This PR adds several missing 3C definitions and relevant overloads

### CODE REVIEW

File: src/Applicative.ts [hunk 1]
```diff
@@ -67,6 +67,13 @@ export interface Applicative3<F extends URIS3> extends Apply3<F> {
   readonly of: <R, E, A>(a: A) => Kind3<F, R, E, A>
 }
 
+/**
+ * @since 2.0.0
```
Comment (gcanti) [inline src/Applicative.ts]: 2.2.0

File: src/Profunctor.ts [hunk 2]
```diff
@@ -31,6 +31,20 @@ export interface Profunctor3<F extends URIS3> extends Functor3<F> {
...
+ * @since 2.2.0
+ */
+export interface Profunctor3C<F extends URIS3, E> extends Functor3C<F, E> {
+  readonly promap: <R, A, D, B>(fbc: Kind3<F, R, E, A>, f: (d: D) => E, g: (a: A) => B) => Kind3<F, R, D, B>
+}
+
+/**
+ * @since 2.2.0
+ */
+export interface Profunctor3C<F extends URIS3, E> extends Functor3C<F, E> {
```
Comment (gcanti) [inline src/Profunctor.ts]: duplicated?

File: src/Applicative.ts (hunk 1 above)
Comment (sledorze) [reply src/Applicative.ts]: ahaha was the main point of the PR :)

File: src/Profunctor.ts (hunk 2 above)
Comment (sledorze) [reply src/Profunctor.ts]: oh! good catch

File: src/Semigroupoid.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export interface Semigroupoid2<F extends URIS2> {
  */
 export interface Semigroupoid2C<F extends URIS2, E> {
   readonly URI: F
+  readonly _E: E
```
Comment (gcanti) [inline src/Semigroupoid.ts]: good catch!

Comment (sledorze) [reply src/Semigroupoid.ts]: sometimes monkeys write books :)

**Comment by gcanti** [comment]
Thanks @sledorze

**Comment by sledorze** [comment]
Thanks @gcanti you (climb the) rock(s)!

**Pull request merged**